aslam_cv_cameras/src/undistorter-mapped.cc

```diff
@@ -0,0 +1,108 @@
+      static_cast<int>(scale * input_camera->imageHeight());
+  PinholeCamera::Ptr output_camera = aslam::createCamera<aslam::PinholeCamera>(
+      intrinsics, output_width, output_height);
+  CHECK(output_camera);
+
+  cv::Mat map_u, map_v;
+  VLOG(1) << "Building undistorter map";
+  common::buildUndistortMap(
+      *input_camera, *output_camera, CV_16SC2, map_u, map_v);
+
+  // Convert map to non-fixed point representation for easy lookup of values.
+  cv::Mat map_u_float = map_u.clone();
+  cv::Mat map_v_float = map_v.clone();
+  aslam::convertMapsLegacy(map_u, map_v, map_u_float, map_v_float, CV_32FC1);
+
+  return std::shared_ptr<MappedUndistorter>(
+      new MappedUndistorter(
+          input_camera, output_camera, map_u, map_v, map_u_float, map_v_float,
+          interpolation_type));
+}
+
+MappedUndistorter::MappedUndistorter()
+    : interpolation_method_(aslam::InterpolationMethod::Linear) {}
+
+MappedUndistorter::MappedUndistorter(
+    Camera::Ptr input_camera, Camera::Ptr output_camera, const cv::Mat& map_u,
+    const cv::Mat& map_v, const cv::Mat& map_u_float,
+    const cv::Mat& map_v_float, aslam::InterpolationMethod interpolation)
+    : Undistorter(input_camera, output_camera),
+      map_u_(map_u),
+      map_v_(map_v),
+      map_u_float_(map_u_float),
+      map_v_float_(map_v_float),
+      interpolation_method_(interpolation) {
+  CHECK_EQ(static_cast<size_t>(map_u_.rows), output_camera->imageHeight());
+  CHECK_EQ(static_cast<size_t>(map_u_.cols), output_camera->imageWidth());
+  CHECK_EQ(static_cast<size_t>(map_v_.rows), output_camera->imageHeight());
+  CHECK_EQ(static_cast<size_t>(map_v_.cols), output_camera->imageWidth());
+  CHECK_EQ(
+      static_cast<size_t>(map_u_float_.rows), output_camera->imageHeight());
+  CHECK_EQ(static_cast<size_t>(map_u_float_.cols), output_camera->imageWidth());
+  CHECK_EQ(
+      static_cast<size_t>(map_v_float_.rows), output_camera->imageHeight());
+  CHECK_EQ(static_cast<size_t>(map_v_float_.cols), output_camera->imageWidth());
+}
+
+void MappedUndistorter::processImage(
+    const cv::Mat& input_image, cv::Mat* output_image) const {
+  CHECK_EQ(input_camera_->imageWidth(), static_cast<size_t>(input_image.cols));
+  CHECK_EQ(input_camera_->imageHeight(), static_cast<size_t>(input_image.rows));
+  CHECK_NOTNULL(output_image);
+  cv::remap(
+      input_image, *output_image, map_u_, map_v_,
+      static_cast<int>(interpolation_method_));
+}
+
+void MappedUndistorter::processPoint(
+    const Eigen::Vector2d& input_point, Eigen::Vector2d* output_point) const {
+  CHECK_NOTNULL(output_point);
+  *output_point = input_point;
+  processPoint(output_point);
+}
+
+void MappedUndistorter::processPoint(Eigen::Vector2d* point) const {
+  VLOG(250) << "Distorted point: " << (*point)[0] << "/" << (*point)[1];
+  CHECK_LE((*point)[0], map_u_.cols);
+  CHECK_LE((*point)[1], map_v_.rows);
+  *point = Eigen::Vector2d(
+      map_u_float_.at<float>((*point)[1], (*point)[0]),
```

Comment:
this seems to be the only reason we have the float maps now, right? Would it be that expensive to do this using the fixed point map?